lib/lint.js

```diff
@@ -12,34 +11,34 @@ const EMPTY_REPORT = {
 	]
 };
 
+async function getCWD(base) {
+	const pkgPath = await getPackagePath(base);
+	return pkgPath ? path.dirname(pkgPath) : base;
+}
+
 // (base: string) => depends: Promise<Boolean>
 async function dependsOnXO(base) {
 	const {dependencies = {}, devDependencies = {}} = await getPackageData(base);
 	return 'xo' in dependencies || 'xo' in devDependencies;
 }
 
-// (editor: Object) => function
-function lint(editor) {
-	const filename = editor.getPath();
-
+// (filename: string) => function
+function lint(filename) {
```

Comment:
The `Task` based lint function won't be able to receive an `editor` object, so I've changed this to only accept the `filename`. This was the only usage of the `editor` argument